src/main/io/gps.c

```diff
@@ -878,30 +1364,40 @@ void gpsUpdate(timeUs_t currentTimeUs)
 
         case GPS_STATE_LOST_COMMUNICATION:
             gpsData.timeouts++;
-            if (gpsConfig()->autoBaud) {
+            // commented out, as it is dangerous and should not be used, fix for a correct auto baud in another pr
+            /*if (gpsConfig()->autoBaud) {
                 // try another rate
                 gpsData.baudrateIndex++;
                 gpsData.baudrateIndex %= GPS_INIT_ENTRIES;
-            }
+            }*/
             gpsSol.numSat = 0;
             DISABLE_STATE(GPS_FIX);
             gpsSetState(GPS_STATE_INITIALIZING);
             break;
 
         case GPS_STATE_RECEIVING_DATA:
             // check for no data/gps timeout/cable disconnection etc
-            if (millis() - gpsData.lastMessage > GPS_TIMEOUT) {
+            DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 2, millis() - gpsData.lastMessage);
+            if (millis() - gpsData.lastMessage > GPS_TIMEOUT_MS(gpsData.updateRate)) {
```

Comment:
cmp32

Reply:
I'm not sure I understand this logic here - why is timeout different based on rate? 2.5s seems like a fair time to think we lost connection, no matter the update rate. This feels unnecessarily complicated

Reply:
Also, you are calculating `millis() - gpsData.lastMessage` twice, both on 1380 and 1381